### PR TITLE
ENG-6188 - ClassName fix for xcode 15 and double event suppression

### PR DESCRIPTION
**Pull request by cometman**

### CODE REVIEW

File: NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift [hunk 1]
```diff
@@ -45,24 +45,28 @@ public extension UIViewController {
...
+        if self.nidClassName == "RNSNavigationController" { return nil }
+        if self.nidClassName == "RNSScreen" { return nil}
+        if self is UINavigationController, nidClassName == "UINavigationController" { return nil }
+        if let tracker = NeuroID.trackers[nidClassName] {
             tracker.subscribe(inScreen: self)
             return tracker
         } else {
-            let tracker = NeuroID.trackers[className] ?? NeuroIDTracker(screen: neuroScreenName, controller: self)
-            NeuroID.trackers[className] = tracker
+            print("NEW TRACKER CREATED!!", nidClassName)
```
Comment (kevinsites) [inline NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:61]: Do we need this line? If so can we put it in our Debug print method?

File: NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift [hunk 2]
```diff
@@ -45,24 +45,28 @@ public extension UIViewController {
...
 
     var tracker: NeuroIDTracker? {
-        if ignoreLists.contains(className) { return nil }
-        if self is UINavigationController, className == "UINavigationController" { return nil }
-        if let tracker = NeuroID.trackers[className] {
+        if ignoreLists.contains(nidClassName) { return nil }
+        // Separating these into individual checks for easy debugging or reverting
+        if self.nidClassName == "RNSNavigationController" { return nil }
+        if self.nidClassName == "RNSScreen" { return nil}
+        if self is UINavigationController, nidClassName == "UINavigationController" { return nil }
```
Comment (kevinsites) [inline NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:56]: Is this the fix for the double registration?

Comment (cometman) [reply NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:56]: @kevinsites yes

**Review by quacke-nid — APPROVED** [review]

File: NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift [hunk 3]
```diff
@@ -143,32 +145,34 @@ internal extension UIViewController {
...
+
         if NeuroID.isStopped() {
             return
         }
 
         // We need to init the tracker on the views.
         tracker
 //        let subViews = view.subviews
-        var allViewControllers = children
+        var allViewControllers = children.filter { !ignoreLists.contains($0.nidClassName) }
```
Comment (kishisaka-nid) [inline NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:157]: This is the trick. We just filter out all the extraneous classnames here.

File: NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift [hunk 4]
```diff
@@ -143,32 +145,34 @@ internal extension UIViewController {
     @objc func neuroIDViewDidAppear() {
         neuroIDViewDidAppear()
 
+        if ignoreLists.contains(nidClassName) { return }
```
Comment (kishisaka-nid) [inline NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:148]: this is the duplicate check?

**Review by kishisaka-nid — APPROVED** [review]
Didn't test it but the code LGTM. 🚢

**Review by priya-xavier — APPROVED** [review]

**Comment by kishisaka-nid** [comment]
suggest we build a test for this later.

File: NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift (hunk 4 above)
Comment (kevinsites) [reply NeuroID/TrackerClassExtensions/UIClassExtensions/UIViewController.swift:148]: Yes and no, we do a similar check on LN 157, but that is checking the children of the current view, this checks the current view itself.

**Pull request merged**